src/plugins/karma.rs

```diff
@@ -47,10 +46,89 @@ pub struct KarmaPlugin {
     re: Regex,
 }
 
+enum ParseState {
+    Nothing,
+
+    ChangeToAdd,
+    Add,
+
+    ChangeToSubtract,
+    Subtract,
+}
+
+/// Parse an input karma change string and return the resulting delta
+fn parse_karma_change(change_str: &str) -> Result<i32> {
+    let mut state = ParseState::Nothing;
+    let mut counter = 0;
+
+    let missing_successive_plus = "you need at least two successive \"+\"s to increment, silly!";
+    let missing_successive_minus = "you need at least two successive \"-\"s to decrement, silly!";
+
+    for c in change_str.chars() {
+        match c {
```

Comment:
The only thing I would personally change is this:

```
match c, state {
    ('+', ParseState::Nothing) => {
        ...
    }
    (unknown_char, _) => {...}
}
```

What are your opinions on that?